tqdm/_tqdm.py

```diff
@@ -751,6 +760,30 @@ def __init__(self, iterable=None, desc=None, total=None, leave=True,
         if file is None:
             file = sys.stderr
 
+        is_std = file in (
```

Comment:
Flash forward, updates coming that will probably be liked by all...  now back to the past and the direct response, in case anybody wants to read it.  Thanks @Julian for knocking me straight again.  I had reasons but the various issues using `io.TextIOWrapper` should have clued me into this being the wrong way to do it.

---

That'd be an option to consider if it didn't raise an exception.

```python
AttributeError: 'file' object has no attribute 'readable'
```

There are other things that `io.TextIOWrapper` also wants that `sys.stderr` does not provide.  That's how `TextIOWrappableStdOutErr` came to be.  Perhaps I should look the other way and instead of adding another layer of wrapping, just not use `TextIOWrapper` at all.  Rather, make my own wrapper thing.

I believe some conversation was had somewhere at some time...  and the expectation was that if `file` were passed then it would need to be a unicode-accepting thing.  If not passed then we use stderr and write bytes.  That makes it a pain to output to stdout so I adjusted to only wrapping stdout/stderr'ish things.  Perhaps instead there should be a separate option to force wrapping of bytes-expecting files.  And/or expose the wrapper class.

I'm going to have to dig a bit more to figure out which cases `not_unicode` was added for.  It may just have been an effort to avoid...  I dunno.